objects/src/accounts/delta/storage.rs

```diff
@@ -73,6 +74,22 @@ impl AccountStorageDelta {
             }
         }
 
+        // make sure storage map deltas are valid
+        for (index, storage_map_delta) in self.updated_maps.iter() {
+            if index > &MAX_MUTABLE_STORAGE_SLOT_IDX {
+                return Err(AccountDeltaError::ImmutableStorageSlot(*index as usize));
+            }
+            // for every storage map delta there should be one corresponding storage item update
+            if !self.updated_items.iter().any(|(idx, _)| idx == index) {
```

Comment:
I'm not sure why we validate this, but doesn't this always yield true? each entry will at least match with itself, wouldn't it?

Reply:
So what I wanted to check here is that every `updated_maps` has a corresponding `updated_items` item. That is because, when we change a map in MASM, the root of the map changes which is stored in a storage item.

In this case `index` is the position of the root of the map in the storage items vector. And `idx` iterates over all updated items. So if there is a map change without a root change, the change is invalid. 

So I think this is correct, no?

Reply:
updated_maps and updated_items are populated when different events in the VM are triggered.

Reply:
ah, I misread and thought there was a `self.updated_maps` in both places. My bad 😅